spec/controllers/two_factor_authentication/personal_key_verification_controller_spec.rb

```diff
@@ -41,39 +41,35 @@
 
   describe '#create' do
     context 'when the user enters a valid personal key' do
+      let(:user) { create(:user, :with_phone) }
+      let(:personal_key) { { personal_key: PersonalKeyGenerator.new(user).create } }
+      let(:payload) { { personal_key_form: personal_key } }
       it 'tracks the valid authentication event' do
-        sign_in_before_2fa(create(:user, :with_webauthn, :with_phone, :with_personal_key))
-        personal_key_generated_at = controller.current_user.
-          encrypted_recovery_code_digest_generated_at
+        freeze_time do
+          personal_key
+          sign_in_before_2fa(user)
+          stub_analytics
 
-        form = instance_double(PersonalKeyForm)
-        response = FormResponse.new(
-          success: true, errors: {}, extra: { multi_factor_auth_method: 'personal-key' },
-        )
-        allow(PersonalKeyForm).to receive(:new).
-          with(subject.current_user, 'foo').and_return(form)
-        allow(form).to receive(:submit).and_return(response)
+          analytics_hash = {
+            success: true,
+            errors: {},
+            multi_factor_auth_method: 'personal-key',
+            multi_factor_auth_method_created_at: Time.zone.now,
```

Comment:
Any reason we chose not to continue asserting against the actual value of the personal key?